### PR TITLE
Issue 4169 - UI - Add PF4 charts for server stats

### PR DESCRIPTION
**Pull request by mreynolds389**
Description:  Added charts for current connections (that does NOT use cn=monitor),
              server memory size and CPU usage.

relates: https://github.com/389ds/389-ds-base/issues/4169

### CODE REVIEW

**Comment by droideck** [comment]
Looks really good!
I have only one small concern both code and visual-wise.

When a user first opens the charts it shows 10 empty values which, I think, looks odd because then it starts back from 1 tick and continues above 10 - so it's a bit confusing.
Can we maybe start from 1 tick to begin with? The chart will optimize it (look at CPU chart I modified). It looks a bit empty but I think that's the point of 'beginning of the monitoring' action:
charts1

An that's how many ticks look like (so everything is okay if we start from 1 and not from 10):
charts2

Also, code-wise, I think these lists can be initialized with something like this:
console.log(Array.from(Array(10).keys(), x => ({"name": "", "x": x + 1, "y": 0})));

What do you think?

**Comment by mreynolds389** [comment]
Okay I think I fixed everything you had a concern about.  Note - I could not use the Array() technique because it basically created an array of Objects (and function), so I had to use a for loop to do optimization of the initialization.  Please review...

**Comment by droideck** [comment]
One small issue and it's probably not this PR's fault but:
cache-chart
For the smaller chart (cache in suffix monitoring), it probably should start from 6 instead of 11.

**Review by droideck — APPROVED** [review]
Besides the minor issue - LGTM!

File: src/cockpit/389-console/src/lib/monitor/serverMonitor.jsx [hunk 1]
```diff
@@ -1,201 +1,550 @@
...
+            connChart: [...this.state.initChart],
+        });
+    }
+
+    refreshCharts() {
+        let cmd = [
+            "ps", "-ef", "|", "grep", "slapd-" + this.props.serverId,
+            "|", "grep", "dirsrv"
+        ];
+        cmd = "ps -ef | grep slapd-" + this.props.serverId + " | grep dirsrv";
```
Comment (tmihinto) [inline src/cockpit/389-console/src/lib/monitor/serverMonitor.jsx]: Second assignment to the variable "cmd".
You might want to remove the first one.

**Pull request merged**